tools/jenkins/run_case.py

```diff
@@ -0,0 +1,137 @@
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# This file will be runned by jenkis to run the e2e perf test
+# Params:
+# am_seserver_dddress: master machines's ip  of standalone environment
+# scenario_file: the file which contains test scenarios
+# flink_home: the path of flink
+# inter_nums：the num of every scenario's running, default value is 10
+# wait_minute: interval time of two elections of qps,default value is 10s
+#
+
+
+import sys
+import time
+
+if sys.version_info < (3, 5):
+    print("Python versions prior to 3.5 are not supported.")
+    sys.exit(-1)
+
+from logger import logger
+from utils import run_command
+from restapi_common import get_avg_qps_by_restful_interface
+
+
+def start_server(flink_home):
+    cmd = "bash %s/bin/start_yarn.sh" % flink_home
+    status, output = run_command(cmd)
+    if status == 0:
+        return 0
+    else:
+        return 1
+
+
+def end_server(flink_home):
+    cmd = "bash %s/bin/stop_yarn.sh" % flink_home
+    run_command(cmd)
+
+
+def get_scenarios(scenario_file_name, test_jar):
+    scenario_file = open(scenario_file_name)
+    line = scenario_file.readline()
+    params_name = []
+    scenarios = []
+    scenario_names = []
+    linenum = 0
+    while line:
+        linenum = linenum + 1
+        cmd = ""
+        scenario_name = ""
+        if linenum == 1:
+            params_name = line.split(" ")
+            for index in range(0, len(params_name)):
+                params_name[index] = params_name.get(index).trip()
+            if not params_name.contains("className"):
+                scenario_file.close()
+                return 1, []
+        else:
+            params_value = line.split(" ")
+            for index in len(params_name):
+                param = params_name.get(index)
+                if param == "className":
+                    cmd = "-c %s %s %s" % (params_value.get(index),test_jar,  cmd)
+                else:
+                    cmd = "%s  --%s %s" % (cmd, param, params_value.get(index))
+                scenario_name = "%s_%s" % (scenario_name, param)
+        scenario_names.append(scenario_name[1:])
+        scenarios.append(cmd)
+        line = scenario_file.readline()
+    scenario_file.close()
+    return 0, scenarios, scenario_names
+
+
+def get_avg(values):
+    sumvalues = 0
+    value_num = 0
+    for value in values:
```

Comment:
here we can test whether len(values) is 0, if it's zero, we will return 0 directly, other than, we can use `sum(values) * 1.0 / len(values)` to calculate the average value of the passed in list